Tests/IntegrationTests/Roslyn/RoslynGenTests.cs

```diff
@@ -43,53 +43,87 @@ public void RoslynPlugin_GenerateForValidAnalyzer_Succeeds()
-            AssertExpectedPropertyDefinitionValue("analyzer1.pkgid1.cs.pluginKey", "analyzer1pkgid1", config);
-            AssertExpectedPropertyDefinitionValue("analyzer1.pkgid1.cs.pluginVersion", "1.0.2", config);
+            string fakeRemoteNuGetDir = TestUtils.CreateTestDirectory(this.TestContext, ".fakeRemoteNuGet");
+            IPackageManager fakeRemotePkgMgr = CreatePackageManager(fakeRemoteNuGetDir);
+            IPackage child1 = AddPackage(fakeRemotePkgMgr, "Analyzer.Child1", "1.1.0", analyzer.GetType().Assembly.Location);
+            IPackage child2 = AddPackage(fakeRemotePkgMgr, "Analyzer.Child2", "1.2.0", analyzer.GetType().Assembly.Location);
+            IPackage targetPkg = AddPackage(fakeRemotePkgMgr, "Empty.Parent", "1.0.0", dummyContentFile, child1, child2);
 
-            // Check the contents of the manifest
-            string actualManifestFilePath = jarChecker.AssertFileExists("META-INF\\MANIFEST.MF");
-            string[] actualManifest = File.ReadAllLines(actualManifestFilePath);
-            AssertExpectedManifestValue(WellKnownPluginProperties.Key, "analyzer1pkgid1", actualManifest);
-            AssertExpectedManifestValue("Plugin-Key", "analyzer1pkgid1", actualManifest); // plugin-key should be lowercase and alphanumeric
-            AssertPackagePropertiesInManifest(analyzerPkg, actualManifest);
+            string localPackageDestination = TestUtils.CreateTestDirectory(this.TestContext, ".localpackages");
 
+            // Act
+            NuGetPackageHandler nuGetHandler = new NuGetPackageHandler(fakeRemotePkgMgr.LocalRepository, localPackageDestination, logger);
+            AnalyzerPluginGenerator apg = new AnalyzerPluginGenerator(nuGetHandler, logger);
+            ProcessedArgs args = new ProcessedArgs(targetPkg.Id, targetPkg.Version, "cs", null, false, 
+                true /* generate plugins for dependencies with analyzers*/, outputDir);
+            bool result = apg.Generate(args);
 
-            // Check the rules
-            string actualRuleFilePath = jarChecker.AssertFileExists("." + config.RulesXmlResourcePath);
-            AssertExpectedRulesExist(analyzer, actualRuleFilePath);
+            // Assert
+            Assert.IsTrue(result);
 
-            // Now create another checker to check the contents of the zip file (strict check this time)
-            CheckEmbeddedAnalyzerPayload(jarChecker, "static\\analyzer1.pkgid1.1.0.2.zip",
-                /* zip file contents */
-                "analyzers\\RoslynAnalyzer11.dll");
+            // Expecting one plugin per dependency with analyzers
+            CheckJarGeneratedForPackage(outputDir, analyzer, child1);
+            CheckJarGeneratedForPackage(outputDir, analyzer, child2);
+            AssertJarsGenerated(outputDir, 2);
+        }
+        
+        [TestMethod]
+        public void RoslynPlugin_GenerateForMultiLevelAnalyzers_Succeeds()
```

Comment:
This test is included as it would have caught an incorrect early return in AnalyzerPluginGenerator.Generate() (now fixed).